### PR TITLE
Add Auth0 provider (without MFA support)

### PR DESCRIPTION
**Pull request by noissefnoc**
I make Auth0 provider (but I don't use MFA, so this provider doesn't support MFA.). 

Could you review this PR?

Following steps are what this code does.

1. Get `connection` (scraping)
    * Scraping `client_id` and `connection` mapping file ( `https://cdn.auth0.com/client/<CLIENT_ID>.js` ).
    * This `connection` is used next step.
2. Get session information (official API)
    * Use [Accept Request](https://auth0.com/docs/api/authentication?http#accept-request) API
    * This requires `connection` but Auth0 doesn't provide official API for getting connections.
    * Get `state` and `CSRF` token.
3. Pretend to login from HTML form by username/password (scraping)
    * Use `state` and `CSRF` token getting before step and pretend to login from HTML form by username/password from `https://<TENANT_NAME>.auth0.com/usernamepassword/login` . (NOTE: Auth0 API document describes `<TENANT_NAME>.auth0.com` as `YOUR_DOMAIN` )
    * Auth0 returns callback information.
    * This includes `SAMLResponse`.
4. SSO to Auth0 (official API)
    * Use [IdP-Initiated Single Sign-On (SSO) Flow](https://auth0.com/docs/api/authentication?http#get-metadata) API
    * This requires `connection` and `SAMLResponse` .
    * This returns SAML assertion.

## related issue

* Issue #510: Auth0 support request

### CODE REVIEW

File: saml2aws.go [hunk 1]
```diff
@@ -4,19 +4,19 @@ import (
 	"fmt"
 	"sort"
 
-	"github.com/versent/saml2aws/v2/pkg/provider/browser"
-	"github.com/versent/saml2aws/v2/pkg/provider/netiq"
```
Comment (noissefnoc) [inline saml2aws.go:8]: Sort Imports

**Comment by noissefnoc** [comment]
@wolfeidau Could you tell me what should I do next for merging this PR?

**Comment by wolfeidau** [comment]
@noissefnoc this looks pretty good, can you take a look at the lint errors.

if you run `make lint-fix` to format the code, and `make lint` to get a list of issues, then sort these out that would be amazing.

**Comment by noissefnoc** [comment]
@wolfeidau Thanks for your reply. I’ll fix it.

**Comment by noissefnoc** [comment]
@wolfeidau I fixed in f5be9bf. I ran `make lint-fix` and `make lint` on my local machine. It says no warnings.

```console
$ make lint-fix
--- lint all the things

$ make lint
--- lint all the things
```

**Comment by wolfeidau** [comment]
@noissefnoc thanks for adding tests this makes it easier for me, and future contributors :) 

The linter covers most of my nit picks.

**Pull request merged**